frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx

```diff
@@ -0,0 +1,188 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { 
+  Form, 
+  FormGroup, 
+  TextInput,
+  Button,
+  ValidatedOptions,
+ } from '@patternfly/react-core';
+import { EyeSlashIcon, EyeIcon } from '@patternfly/react-icons';
+import {
+  SecretKind,
+  apiVersionForModel,
+} from '@console/internal/module/k8s';
+import {
+  SecretModel
+} from '@console/internal/models';
+import { isValidUrl } from '@console/shared';
+import { CreatePayload, ExternalComponentProps, CanGoToNextStep } from '../types';
+import { FlashsystemState, IBMFlashsystemKind } from './type';
+import { IBMFlashsystemModel } from './models';
+
+export const FlashsystemConnectionDetails: React.FC<ExternalComponentProps<FlashsystemState>> = ({ 
+  setFormState,
+  formState,
+}) => {
+  const { t } = useTranslation();
+  const [reveal, setReveal] = React.useState(false);
+  const [endpointValid, setEndpointValid] = React.useState(ValidatedOptions.default);
+  const onChange = (value: string) => {
+    setFormState('endpoint', value);
+    if (value){
+      if (isValidUrl(value)){
+        setEndpointValid(ValidatedOptions.success);
+      } else {
+        setEndpointValid(ValidatedOptions.error);
+      }
+    };
+  };
+
+  return (
+    <Form>
+      <FormGroup 
+        label={t('ceph-storage-plugin~Endpoint')} 
+        fieldId="endpoint-input"
+        isRequired
+        validated = {endpointValid}
+        helperText={
+          t('ceph-storage-plugin~Rest API IP address of IBM Storage FlashSystem.')
+        }
+        helperTextInvalid={t('ceph-storage-plugin~The endpoint is not a valid URL')}
+        >
+        <TextInput
+          id="endpoint-input"
+          value={formState.endpoint}
+          type="text"
+          onChange={onChange}
+          isRequired
+        />
+      </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Username')} isRequired fieldId="username-input">
+        <TextInput
+          id="username-input"
+          value={formState.username}
+          type="text"
+          onChange={(value: string) => setFormState('username', value)}
+          isRequired
+        />
+      </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Password')} isRequired fieldId="password-input" >
+      {reveal ? (
+        <>
+        <TextInput
+          id="password-input"
+          value={formState.password}
+          type="text"
+          onChange={(value: string) => setFormState('password', value)}
+          isRequired
+        />
```

Comment:
cant we do something like: 
```suggestion
        <TextInput
          id="password-input"
          value={formState.password}
          type={reveal ? 'text' : 'password'}
          onChange={(value: string) => setFormState('password', value)}
          isRequired
        />
```